src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -183,11 +185,33 @@ public void testForInvalidCheckImplementation() throws Exception {
 
     @Test
     public void testProcessNonJavaFiles() throws Exception {
+        final List<String> lines =
+            new ArrayList<>(Arrays.asList("package com.puppycrawl.tools.checkstyle;", "",
+                "error public class InputTreeWalkerFileWithViolation {}"));
+
         final TreeWalker treeWalker = new TreeWalker();
-        treeWalker.setTabWidth(1);
+        final PackageObjectFactory factory = new PackageObjectFactory(
+            new HashSet<>(), Thread.currentThread().getContextClassLoader());
+        treeWalker.setModuleFactory(factory);
         treeWalker.configure(new DefaultConfiguration("default config"));
-        final File file = new File("src/main/resources/checkstyle_packages.xml");
-        treeWalker.processFiltered(file, new ArrayList<>());
+        final DefaultConfiguration childConfig = createCheckConfig(JavadocParagraphCheck.class);
+        treeWalker.setupChild(childConfig);
+
+        try {
+            treeWalker.processFiltered(new File("input.java"), lines);
+            fail("Exception expected");
+        }
+        catch (CheckstyleException ex) {
+            assertEquals("Invalid exception message",
+                "MismatchedTokenException occurred during the analysis of file input.java.",
+                ex.getMessage());
+        }
+        try {
+            treeWalker.processFiltered(new File("input.txt"), lines);
+        }
+        catch (CheckstyleException ex) {
+            fail("Exception is not expected, file with txt extension should not be processed");
```

Comment:
The situation is as follows: method processFiltered if called from BaseCheckTestSupport [this](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L170) condition is not checked as it checks it before [here](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java#L78).
So I have to call it on TreeWalker class, and check that on the same lines but different extensions it fails on java file and does not on txt. What shoul I pass to junit runner and for what purpose, if the cause of test failure is not in the exception but in the fact that it is thrown?